src/pbt/__init__.py

```diff
@@ -30,104 +43,233 @@ def __init__(self, path_root: str):
+                print("   Create/Update failed for jobs: %s" % (" ,".join(job_update_failures.keys())))
+            sys.exit(1)
+
+    def run_unit_tests(self):
+        if self._verify_unit_test_env():
+            unit_test_results = {}
+
+            for pipeline_i, (path_pipeline, pipeline) in enumerate(self.pipelines.items()):
+                print('\n  Unit Testing pipeline %s [%s/%s]' % (path_pipeline, pipeline_i + 1, self.pipelines_count))
+
+                path_pipeline_absolute = os.path.join(os.path.join(self.path_root, path_pipeline), 'code')
+                if self.project_language == 'python':
+                    if os.path.isfile(os.path.join(path_pipeline_absolute, 'test/TestSuite.py')):
+                        unit_test_results[path_pipeline] = self.test_python(path_pipeline_absolute)
+                elif self.project_language == 'scala':
+                    unit_test_results[path_pipeline] = self.test_scala(path_pipeline_absolute)
+            is_any_ut_failed = False
+            for path_pipeline, return_code in unit_test_results.items():
+                if return_code not in (0, 5):
+                    is_any_ut_failed = True
+                    print(
+                        f"\n[bold red]❌ Unit test for pipeline: {path_pipeline} failed with return code {return_code}![/bold red]")
+                else:
+                    print(f"\n[bold blue]✅ Unit test for pipeline: {path_pipeline} succeeded.[/bold blue]")
+
+            if is_any_ut_failed:
+                sys.exit(1)
+        else:
+            sys.exit(1)
+
+    def build_python(self, path_pipeline_absolute):
+        return self._process_sequential(
+            [
+                # Install dependencies of particular pipeline
+                # 1. Export dependencies to egg_info/requires.txt
+                Process(['python3', 'setup.py', '-q', 'egg_info'], path_pipeline_absolute),
+                # 2. Extract the install_requires, tests_requires and extra_requires and install them
+                Process([f"pip3 install -q `grep -v '^\[' *.egg-info/requires.txt`"],
+                        path_pipeline_absolute, is_shell=True),
+                # Check for compilation errors
+                Process(['python3', '-m', 'compileall', '.', '-q'], path_pipeline_absolute),
+                # Generate wheel
+                Process(['python3', 'setup.py', 'bdist_wheel'], path_pipeline_absolute)
+            ]
+        )
+
+    def build_scala(self, path_pipeline_absolute):
+        return self._process_sequential(
+            [Process(['mvn', 'clean', 'package', '-q', '-DskipTests'], path_pipeline_absolute)])
+
+    def test_scala(self, path_pipeline_absolute):
+        return self._process_sequential(
+            [
+                Process(['mvn', 'test', '-q', '-Dfabric=' + self.fabric.strip()],
+                        path_pipeline_absolute)
+            ]
+        )
+
+    def test_python(self, path_pipeline_absolute):
```

Comment:
Any reason for writing the build, test functions and command in _init_.py? It would be better to create different files for build, test and so on. This content is too big for `__init__.py`

Reply:
Created separate files for each class. So build/test/deploy are inside the pbt class inside a file but in a single file